src/components/collapsible-motion/collapsible-motion.js

```diff
@@ -17,144 +25,116 @@ const createClosingAnimation = height =>
-      {
-        fullHeight: this.nodeRef.current.clientHeight,
-      },
-      this.props.toggle
-    );
-  };
-
-  render() {
-    return this.props.children({
-      animation: this.state.animation,
-      containerStyles: this.state.containerStyles,
-      toggle: this.handleToggle,
-      registerContentNode: this.nodeRef,
-    });
-  }
-}
-
-class CollapsibleMotion extends React.PureComponent {
-  static displayName = 'CollapsibleMotion';
-  static propTypes = {
-    children: PropTypes.func.isRequired,
-    isClosed: PropTypes.bool,
-    onToggle: PropTypes.func,
-    isDefaultClosed: PropTypes.bool,
-  };
-
-  render() {
-    return (
-      <Collapsible
-        isClosed={this.props.isClosed}
-        isDefaultClosed={this.props.isDefaultClosed}
-        onToggle={this.props.onToggle}
-      >
-        {({ isOpen, toggle }) => (
-          <ToggleAnimation isOpen={isOpen} toggle={toggle}>
-            {({
-              animation,
-              containerStyles,
-              toggle: animationToggle,
-              registerContentNode,
-            }) => (
-              <ClassNames>
-                {({ css }) => {
-                  let animationStyle = {};
-
-                  if (animation) {
-                    // By calling `css`, emotion injects the required CSS into the document head.
-                    // eslint-disable-next-line no-unused-expressions
-                    css`
-                      animation: ${animation} 200ms forwards;
-                    `;
-                    animationStyle = {
-                      animation: `${animation.name} 200ms forwards`,
-                    };
-                  }
-
-                  return this.props.children({
-                    isOpen,
-                    containerStyles: {
-                      ...containerStyles,
-                      ...animationStyle,
-                    },
-                    toggle: animationToggle,
-                    registerContentNode,
-                  });
-                }}
-              </ClassNames>
-            )}
-          </ToggleAnimation>
-        )}
-      </Collapsible>
-    );
+    toggle();
+  }, [nodeRef, toggle]);
+
+  const containerStyles = props.isOpen
+    ? { height: 'auto' }
+    : { height: 0, overflow: 'hidden' };
+
+  let animation = null;
+
+  // if state has changed
+  if (typeof prevIsOpen !== 'undefined' && prevIsOpen !== props.isOpen) {
+    animation = props.isOpen
+      ? createOpeningAnimation(nodeRef.current.clientHeight)
+      : createClosingAnimation(nodeRef.current.clientHeight);
   }
-}
+
+  return props.children({
+    animation,
+    containerStyles,
+    toggle: handleToggle,
+    registerContentNode: nodeRef,
+  });
+};
+
+ToggleAnimation.displayName = 'ToggleAnimation';
+ToggleAnimation.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  toggle: PropTypes.func.isRequired,
+  children: PropTypes.func.isRequired,
+};
+
+const CollapsibleMotion = props => {
+  return (
+    <Collapsible
```

Comment:
I want to use `useToggleState` here, but it's a bit complicated because `CollapsibleMotion` can be both uncontrolled and controlled...